### PR TITLE
Changes to support Genesis Plus GX and Stella cores

### LINKED ISSUE #3: `ArrayVideoDriver.screenshot()` doesn't rotate the returned buffer

### LINKED ISSUE #5: `ArrayVideoDriver.screenshot()` does not handle 16-bit pixel formats
`ArrayVideoDriver.screenshot()` does not handle the RGB565 or RGB1555 pixel formats, which are used by a lot of 8-bit system cores.

(I have written code for this, and would like to discuss.)

### LINKED ISSUE #6: Stella core throws error starting a session.
The stella_libretro core throws an error when starting a session:

`Exception ignored on calling ctypes callback function: <bound method DictEnvironmentDriver.environment of <libretro.drivers.environment.composite.CompositeEnvironmentDriver object at 0x0000025FFECE9C10>>
Traceback (most recent call last):
  File "C:\Users\jsens\AppData\Roaming\Python\Python312\site-packages\libretro\drivers\environment\dict.py", line 44, in environment
    return self._envcalls[envcall](data)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\jsens\AppData\Roaming\Python\Python312\site-packages\libretro\drivers\environment\default.py", line 160, in <lambda>
    EnvironmentCall.SET_GEOMETRY: lambda data: self._set_geometry(
                                               ^^^^^^^^^^^^^^^^^^^
  File "C:\Users\jsens\AppData\Roaming\Python\Python312\site-packages\libretro\drivers\environment\composite.py", line 869, in _set_geometry
    self._video.geometry = geometry_ptr[0]
    ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\jsens\AppData\Roaming\Python\Python312\site-packages\libretro\drivers\video\multi.py", line 304, in geometry
    self._system_av_info.geometry.base_width = value.base_width
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'geometry'`

The core appears to run fine after the error.

### LINKED ISSUE #8: Frame buffer can be too small if frame pitch is larger than expected.
Using the Genesis Plus GX core, the pitch can be much larger than the width in pixels times the pixel format width. This leads to an error in `ArrayVideoDriver.refresh()`:

```
Exception ignored on calling ctypes callback function: <bound method CompositeEnvironmentDriver.video_refresh of <libretro.drivers.environment.composite.CompositeEnvironmentDriver object at 0x000001B5E102DC10>>
Traceback (most recent call last):
  File "C:\Users\jsens\AppData\Roaming\Python\Python312\site-packages\libretro\drivers\environment\composite.py", line 332, in video_refresh
    self._video.refresh(view, width, height, pitch)
  File "C:\Users\jsens\AppData\Roaming\Python\Python312\site-packages\libretro\drivers\video\multi.py", line 89, in refresh
    self._current.refresh(data, width, height, pitch)
  File "C:\Users\jsens\AppData\Roaming\Python\Python312\site-packages\libretro\drivers\video\software\array.py", line 33, in refresh
    frameview[: len(data)] = data
    ~~~~~~~~~^^^^^^^^^^^^^
ValueError: memoryview assignment: lvalue and rvalue have different structures
```

Reallocating the buffer when it is too small rectifies the issue.

### PR DESCRIPTION
**Pull request by JSensebe**
Added 16-bit pixel format and rotation support  to ArrayVideoDriver.screenshot(). Added buffer reallocation when necessary to ArrayVideoDriver.refresh(). Added check when setting geometry before av_info is filled in MultiVideoDriver.

- Fixes #3
- Fixes #5
- Fixes #6
- Fixes #8

### CODE REVIEW

File: src/libretro/drivers/video/software/array.py [hunk 1]
```diff
@@ -93,26 +98,91 @@ def pixel_format(self, format: PixelFormat) -> None:
...
+            case Rotation.ONE_EIGHTY:
+                sy = self._last_width * self._last_height * 4 - 4
+                dx = -4
+                dy = self._last_width * -4
+                is_sideways = False
+            case Rotation.TWO_SEVENTY:
+                sy = self._last_height * 4 - 4
+                dx = self._last_height * 4
+                dy = -4
+                is_sideways = True
```
Comment (JesseTG) [inline src/libretro/drivers/video/software/array.py:132]: Doesn't your logic here assume a pixel size of 4 bytes? Or is that just for the output buffer?

File: src/libretro/drivers/video/driver.py [hunk 2]
```diff
@@ -247,7 +247,7 @@ def shared_context(self) -> bool: ...
     def shared_context(self, value: bool) -> None: ...
 
     @abstractmethod
-    def screenshot(self) -> Screenshot | None:
+    def screenshot(self, prerotate: bool = True) -> Screenshot | None:
```
Comment (JesseTG) [inline src/libretro/drivers/video/driver.py:250]: Could you add details about this parameter to the doc comment, please?

**Review by JesseTG — CHANGES_REQUESTED** [review]
Looks good overall, just a few things I'd appreciate some clarity on.

File: src/libretro/drivers/video/software/array.py (hunk 1 above)
Comment (JSensebe) [reply src/libretro/drivers/video/software/array.py:132]: It's just the output buffer. I'll add some comments to make the code clearer.

**Pull request merged**